Assets/Scripts/ResourceManager.cs

```diff
@@ -44,6 +44,8 @@ void UpdateUI()
             + "Time elapsed: " + timeElapsedSec.ToString("0.0") + "\n"
             + "Time since last hit: " + GetTimeSinceLastHit().ToString("0.0") + "\n"
             + "Current State: " + k_move.movementState + "\n"
+            + "RB constraints: " + k_move.GetRigidbodyConstraints() + "\n"
+            + "Hit input: " + Input.GetAxis("Jump") + "\n"
```

Comment:
Just adding some potentially relevant debug fields to the UI. Didn't end up being the issue for me, but could be helpful in the future.